addons/project/models/res_partner.py

```diff
@@ -61,3 +61,24 @@ def _create_portal_users(self):
                 'active': True,
             })
         return created_users
+
+    def action_view_tasks(self):
+        self.ensure_one()
+        action = {
+            **self.env["ir.actions.actions"]._for_xml_id("project.project_task_action_from_partner"),
+            'display_name': _("%(partner_name)s's Tasks", partner_name=self.name),
+            'context': {
+                'default_partner_id': self.id,
+            },
+        }
+        all_child = self.with_context(active_test=False).search([('id', 'child_of', self.ids)])
+        search_domain = [('partner_id', 'in', (self | all_child).ids)]
+        if self.task_count <= 1:
+            task_id = self.env['project.task'].search(search_domain, limit=1)
+            action['res_id'] = task_id.id
+            action['views'] = [
+                (self.env['ir.model.data']._xmlid_to_res_id('project.view_task_form_res_partner'), 'form'),
+            ]
```

Comment:
I am not sure but I guess the form view is already in `views` of the action. We could maybe just filter the array to only have the form view as element inside that array. 🤷🏼‍♂️ 

```suggestion
            action['views'] = [(view_id, view_type) for view_id, view_type if view_type == "form"]
```